modules/swagger-codegen/src/main/resources/cpprest/api-source.mustache

```diff
@@ -53,7 +53,22 @@ pplx::task<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}void{{/r
     utility::string_t responseHttpContentType;
     
     // use JSON if possible
-    if ( responseHttpContentTypes.size() == 0 || responseHttpContentTypes.find(U("application/json")) != responseHttpContentTypes.end() )
+    if ( responseHttpContentTypes.size() == 0 )
+    {
+        {{#vendorExtensions.x-codegen-response.isPrimitiveType}}
```

Comment:
Instead of using {{#vendorExtensions.x-codegen-response.isPrimitiveType}, what about using `{{#primitiveType}}` in the response object instead?

https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/CodegenResponse.java#L15